src/data_processing/data_processing_and_training_demo.ipynb

```diff
@@ -0,0 +1,1559 @@
+      "Epoch: 37/40, Step: 900/1575, Loss: 0.0059\n",
+      "Epoch: 37/40, Step: 1000/1575, Loss: 0.0060\n",
+      "Epoch: 37/40, Step: 1100/1575, Loss: 0.0055\n",
+      "Epoch: 37/40, Step: 1200/1575, Loss: 0.0054\n",
+      "Epoch: 37/40, Step: 1300/1575, Loss: 0.0049\n",
+      "Epoch: 37/40, Step: 1400/1575, Loss: 0.0069\n",
+      "Epoch: 37/40, Step: 1500/1575, Loss: 0.0056\n",
+      "Epoch: 38/40, Step: 100/1575, Loss: 0.0063\n",
+      "Epoch: 38/40, Step: 200/1575, Loss: 0.0062\n",
+      "Epoch: 38/40, Step: 300/1575, Loss: 0.0050\n",
+      "Epoch: 38/40, Step: 400/1575, Loss: 0.0069\n",
+      "Epoch: 38/40, Step: 500/1575, Loss: 0.0063\n",
+      "Epoch: 38/40, Step: 600/1575, Loss: 0.0051\n",
+      "Epoch: 38/40, Step: 700/1575, Loss: 0.0069\n",
+      "Epoch: 38/40, Step: 800/1575, Loss: 0.0069\n",
+      "Epoch: 38/40, Step: 900/1575, Loss: 0.0048\n",
+      "Epoch: 38/40, Step: 1000/1575, Loss: 0.0058\n",
+      "Epoch: 38/40, Step: 1100/1575, Loss: 0.0064\n",
+      "Epoch: 38/40, Step: 1200/1575, Loss: 0.0050\n",
+      "Epoch: 38/40, Step: 1300/1575, Loss: 0.0059\n",
+      "Epoch: 38/40, Step: 1400/1575, Loss: 0.0059\n",
+      "Epoch: 38/40, Step: 1500/1575, Loss: 0.0055\n",
+      "Epoch: 39/40, Step: 100/1575, Loss: 0.0050\n",
+      "Epoch: 39/40, Step: 200/1575, Loss: 0.0066\n",
+      "Epoch: 39/40, Step: 300/1575, Loss: 0.0055\n",
+      "Epoch: 39/40, Step: 400/1575, Loss: 0.0051\n",
+      "Epoch: 39/40, Step: 500/1575, Loss: 0.0055\n",
+      "Epoch: 39/40, Step: 600/1575, Loss: 0.0057\n",
+      "Epoch: 39/40, Step: 700/1575, Loss: 0.0055\n",
+      "Epoch: 39/40, Step: 800/1575, Loss: 0.0052\n",
+      "Epoch: 39/40, Step: 900/1575, Loss: 0.0058\n",
+      "Epoch: 39/40, Step: 1000/1575, Loss: 0.0062\n",
+      "Epoch: 39/40, Step: 1100/1575, Loss: 0.0057\n",
+      "Epoch: 39/40, Step: 1200/1575, Loss: 0.0064\n",
+      "Epoch: 39/40, Step: 1300/1575, Loss: 0.0053\n",
+      "Epoch: 39/40, Step: 1400/1575, Loss: 0.0055\n",
+      "Epoch: 39/40, Step: 1500/1575, Loss: 0.0062\n",
+      "Epoch: 40/40, Step: 100/1575, Loss: 0.0064\n",
+      "Epoch: 40/40, Step: 200/1575, Loss: 0.0072\n",
+      "Epoch: 40/40, Step: 300/1575, Loss: 0.0056\n",
+      "Epoch: 40/40, Step: 400/1575, Loss: 0.0049\n",
+      "Epoch: 40/40, Step: 500/1575, Loss: 0.0059\n",
+      "Epoch: 40/40, Step: 600/1575, Loss: 0.0072\n",
+      "Epoch: 40/40, Step: 700/1575, Loss: 0.0055\n",
+      "Epoch: 40/40, Step: 800/1575, Loss: 0.0063\n",
+      "Epoch: 40/40, Step: 900/1575, Loss: 0.0056\n",
+      "Epoch: 40/40, Step: 1000/1575, Loss: 0.0057\n",
+      "Epoch: 40/40, Step: 1100/1575, Loss: 0.0058\n",
+      "Epoch: 40/40, Step: 1200/1575, Loss: 0.0055\n",
+      "Epoch: 40/40, Step: 1300/1575, Loss: 0.0054\n",
+      "Epoch: 40/40, Step: 1400/1575, Loss: 0.0052\n",
+      "Epoch: 40/40, Step: 1500/1575, Loss: 0.0065\n"
+     ]
+    }
+   ],
+   "source": [
+    "#use dataloader to train the model\n",

```

Comment:
also @soapath, i think most of the training script is done here so u can just copy it and adjust it however you would like (by saving the model into a models/ folder every epoch or something when u write a training script)